Real Analysis/real-analysis-notes.tex

```diff
@@ -573,4 +576,371 @@ \subsection{Measurable spaces}
+& = \mu(E_1) - \mu\left( \bigcup_{n \geq 1} K_n \right) & \text{
+                                                          because
+                                                          these sets
+                                                          are
+                                                          finite.}\\
+& = \mu(E_1) - \lim_{n \to \infty} \mu(K_n) & \text{by continuity from
+                                              below.} \\
+& = \lim_{n \geq \infty} \left[ \mu(E_1) - \mu(K_n) \right] & \text{
+                                                              because
+                                                              }K_n\text{
+                                                              is a
+                                                              bounded
+                                                              sequence.}\\
+      & = \lim_{n \geq \infty} E_n&
+    \end{align*}
+
+  \end{enumerate}
+\end{proof}
+\begin{rmk}
+  With the definition of + on $[0,\infty)$, we have $a+c = b+c \
+  \implies a=b$ if $c \neq \infty$. Otherwise, this need not be true.
+\end{rmk}
+\begin{defn}
+  $E \in \M$ is called a \de{null set} if and only if $\mu(E) = 0$.
+\end{defn}
+\begin{defn}
+  Suppose $P(x)$ is a logical predicate dependent on $x \in X \in (X,
+  \M, \mu)$. We say $P(x)$ is true \de{$\mu$-almost everywhere} if and
+  only if there exists $E \in \M$ a null set such that $\neg P(x) 
+  \implies  x \in E$. 
+\end{defn}
+\begin{rmk}
+  We note that $\{x \in X \mid P(x) \text{ is false}\} \subset E$ with
+  $E \subset \M$ such that $\mu(E) = 0$. 
+\end{rmk}
+\begin{defn}
+  A measure space $(X,\M,\mu)$ is called a \de{complete measure space}
+  if and only if every subset of a null set is contained in $\M$. 
+\end{defn}
+\begin{rmk}
+  This implies that the measure of any such set would be 0.
+\end{rmk}
+\begin{thm}
+  Let $(X,\M,\mu)$ be a measure space. Furthermore, let \[
+    \ov{\M} = \{E \cup F \mid E \in \M, \exists Z \in \M \text{ with }
+    \mu(Z) = 0 \text{ and } F \subset Z\}.
+  \]
+  Then, $\ov{M}$ is the $\sigma$-algebra on $X$ containing $\M$ and
+  there exists a unique $\ov{\mu}$ extending $\mu$ from $\M \to
+  \ov{M}$ such that $(X,\ov{M},\ov{\mu})$ is complete.
+\end{thm}
+\begin{proof}
+  Given such a complete space, we get that $\emptyset = \emptyset \cup
+  \emptyset$. Since $\emptyset \in \M$  and $\emptyset \subset
+  \emptyset \subset \M$ with $\mu(\emptyset) = 0$, we get $\emptyset
+  \in \ov{M}$. \\
+
+  Now, let $E \cup F \in \M$ where $F \subset Z$ with $Z$
+  null. Then,
+  \begin{align*}
+    (E \cup F)^c & = E^c \cap F^c \\
+                 & = E^c \cap \left( (F^c \setminus Z^c) \cup Z^c \right) \\
+    & = \left( E^c \cap (F^c \setminus Z^c) \right) \cup (E \cap Z^c)
+    \\
+    & = \left( E^c \cap (Z \setminus F) \right) \cup (E \cap Z^c)
+  \end{align*}
+  and since $Z \setminus F \subset Z$, we are done.
+\end{proof}
```

Comment:
maybe add a remark saying that using the idea of the null sets one may extend the measure beyond the sets it is originally defined?